scenarios/aws/microVMs/microvms/libvirt.go

```diff
@@ -82,14 +82,19 @@ type VMCollection struct {
 
 func (vm *VMCollection) SetupCollectionFilesystems(depends []pulumi.Resource) ([]pulumi.Resource, error) {
 	var waitFor []pulumi.Resource
+	var err error
 
+	var libvirtPoolReady []pulumi.Resource
 	if vm.instance.Arch != LocalVMSet {
-		libvirtPoolReady, err := setupRemoteLibvirtPool(vm.pool, vm.instance.runner, depends)
-		if err != nil {
-			return []pulumi.Resource{}, err
-		}
-		depends = append(depends, libvirtPoolReady...)
+		libvirtPoolReady, err = setupRemoteLibvirtPool(vm.pool, vm.instance.runner, depends)
+	} else {
+		libvirtPoolReady, err = setupLocalLibvirtPool(vm.instance.e.Ctx, vm.libvirtProvider, vm.pool, depends)
+
+	}
+	if err != nil {
+		return []pulumi.Resource{}, err
```

Comment:
why not `return nil, err` ?

Reply:
No good reason I guess. I think I misunderstood that empty slice == nil in pretty much all aspects in golang.

I would like to keep these as-is for now. And then make a seperate PR to change all instances of this behavior in the code, for consistency.